### PR TITLE
feat: add drivers page BE

### PR DESCRIPTION
**Pull request by nidalhjab**
- Add drivers page backend.

### CODE REVIEW

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+export const getDrivers = async (req: Request, res: Response) => {
+	const userId = req.params.id;
+	if (userId) {
+		const response = await service.getUserDrivers(req, userId);
+		return res.send(response);
+	} else {
+		return res.send({ error: "Invalid input" });
+	}
+};
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:12]: always handle error first

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	const phone: string = req.body.phone;
+	const carModel: string = req.body.carModel;
+	const licensePlate: string = req.body.licensePlate;
+	const date: Date = req.body.date;
+	const locationId: number = req.body.locationId;
+	const userId: number = req.body.userId;
+
+	if (!name || !phone || !carModel || !licensePlate || !date || !locationId || !userId) {
+		return res.send({ error: "Invalid input" });
+	}
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:26]: we need a proper error messages with the missing fields

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	const phone: number = req.body.phone;
+	const carModel: string = req.body.carModel;
+	const licensePlate: string = req.body.licensePlate;
+	const date: Date = req.body.date;
+	const locationId: number = req.body.locationId;
+	const userId: number = req.body.userId;
+
+	if (!name || !phone || !carModel || !licensePlate || !date || !locationId || !userId) {
+		return res.send({ error: "Invalid input" });
+	}
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:67]: same as above

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+export const putDriver = async (req: Request, res: Response) => {
+	const name: string = req.body.name;
+	let lat: number = req.body.lat;
+	let lng: number = req.body.lng;
+	const phone: number = req.body.phone;
+	const carModel: string = req.body.carModel;
+	const licensePlate: string = req.body.licensePlate;
+	const date: Date = req.body.date;
+	const locationId: number = req.body.locationId;
+	const userId: number = req.body.userId;
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:63]: destruct the values.

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+export const postDriver = async (req: Request, res: Response) => {
+	const name: string = req.body.name;
+	let lat: number = req.body.lat;
+	let lng: number = req.body.lng;
+	const phone: string = req.body.phone;
+	const carModel: string = req.body.carModel;
+	const licensePlate: string = req.body.licensePlate;
+	const date: Date = req.body.date;
+	const locationId: number = req.body.locationId;
+	const userId: number = req.body.userId;
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:22]: destruct the values

File: packages/server/src/routes/drivers/driver-routes.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import * as express from "express";
+import { getDrivers, postDriver, deleteDriver, putDriver } from "../../controllers/driver/driver-controller";
+
+const router = express.Router();
+
+router.get("/:id", getDrivers);
+router.post("/", postDriver);
+router.delete("/:id", deleteDriver);
+router.put("/", putDriver);
```
Comment (hamzahmurrar) [inline packages/server/src/routes/drivers/driver-routes.ts:9]: we dont have getDrivers to get them all ?

**Review by hamzahmurrar — CHANGES_REQUESTED** [review]
1- I left a couple of comments -> lets stick with code consistency.
2- handle errors first.
3- return proper error messages.
4- add a middleware for authentication not on each route.

File: packages/server/src/routes/drivers/driver-routes.ts (hunk 6 above)
Comment (nidalhjab) [reply packages/server/src/routes/drivers/driver-routes.ts:9]: Every user gets its related drivers(the drivers that the user adds). @hamzahmurrar

File: packages/server/src/controllers/driver/driver-controller.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	if (!date) {
+		return res.send({ error: "Date is required" });
+	}
+	if (!locationId) {
+		return res.send({ error: "Location is required" });
+	}
+	if (!userId) {
+		return res.send({ error: "User-Id is required" });
+	}
```
Comment (hamzahmurrar) [inline packages/server/src/controllers/driver/driver-controller.ts:35]: we need a status code here + one return statement for all the errors.

**Pull request merged**